Release-checklist.md

```diff
@@ -0,0 +1,37 @@
+# Pre-release
+* triage newer bugs and open PRs (check the milestone if one exists)
+* fix, revert and delay the release or merges if necessary
+* test a clean build (handled by travis)
+
+## Runtime check
+* IMP creation
+* AIM and extending contracts
+* taking Drassen airport or more
+* buying from Bobby Ray's
+* visual inspection of laptop screens
+
+# Release
+* update changes.md
+  * update man page if necessary
+* bump version
+* github release + tags
+
+## Build releases for distribution
+###  Windows release on Linux: 
```

Comment:
This was just moved from the Makefile. It should eventually mention all the release targets (BSD notably missing).